src/Build.UnitTests/Construction/SolutionFile_Tests.cs

```diff
@@ -1913,5 +1913,339 @@ public void ParseProjectConfigurationsInSolutionConfigurations2()
             Assert.Equal(".NET", solution.GetDefaultPlatformName()); // "Default solution platform"
         }
+
+        [Fact]
+        public void ParseSolutionFileContainingProjectsWithSimilarNames_TwoProjects()
+        {
+            string solutionFileContents =
+                @"
+                Microsoft Visual Studio Solution File, Format Version 12.00
+                # Visual Studio 15
+                VisualStudioVersion = 15.0.27130.2010
+                MinimumVisualStudioVersion = 10.0.40219.1
+                Project('{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}') = 'Project.Named.With.Dots', 'Project.Named.With.Dots.csproj', '{FC2889D9-6050-4D2E-B022-979CCFEEAAAC}'
+                EndProject
+                Project('{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}') = 'Project_Named_With_Dots', 'Project_Named_With_Dots.csproj', '{ED30D4A3-1214-410B-82BB-B61E5A9D05CA}'
+                EndProject
+                Global
+	                GlobalSection(SolutionConfigurationPlatforms) = preSolution
+		                Release|Any CPU = Release|Any CPU
+	                EndGlobalSection
+	                GlobalSection(ProjectConfigurationPlatforms) = postSolution
+		                {FC2889D9-6050-4D2E-B022-979CCFEEAAAC}.Release|Any CPU.ActiveCfg = Release|Any CPU
+		                {FC2889D9-6050-4D2E-B022-979CCFEEAAAC}.Release|Any CPU.Build.0 = Release|Any CPU
+		                {ED30D4A3-1214-410B-82BB-B61E5A9D05CA}.Release|Any CPU.ActiveCfg = Release|Any CPU
+		                {ED30D4A3-1214-410B-82BB-B61E5A9D05CA}.Release|Any CPU.Build.0 = Release|Any CPU
+	                EndGlobalSection
+	                GlobalSection(SolutionProperties) = preSolution
+		                HideSolutionNode = FALSE
+	                EndGlobalSection
+	                GlobalSection(ExtensibilityGlobals) = postSolution
+		                SolutionGuid = {C038ED6B-BFC1-4E50-AE2E-7993F6883D7F}
+	                EndGlobalSection
+                EndGlobal
+                ";
+
+            SolutionFile solution = ParseSolutionHelper(solutionFileContents);
+
+            ProjectInSolution project1 = (ProjectInSolution)solution.ProjectsByGuid["{FC2889D9-6050-4D2E-B022-979CCFEEAAAC}"];
+            ProjectInSolution project2 = (ProjectInSolution)solution.ProjectsByGuid["{ED30D4A3-1214-410B-82BB-B61E5A9D05CA}"];
+
+            Assert.NotEqual(project1.GetUniqueProjectName(), project2.GetUniqueProjectName());
+            Assert.Equal("Project_Named_With_Dots_{FC2889D9-6050-4D2E-B022-979CCFEEAAAC}", project1.GetUniqueProjectName());
+            Assert.Equal("Project_Named_With_Dots", project2.GetUniqueProjectName());
+
+            Assert.Equal("Project.Named.With.Dots", project1.GetOriginalProjectName());
+            Assert.Equal("Project_Named_With_Dots", project2.GetOriginalProjectName());
+        }
+
+        [Fact]
+        public void ParseSolutionFileContainingProjectsWithSimilarNames_TwoProjects_DifferentOrder()
```

Comment:
Can you collapse these two tests into a single [Theory](https://xunit.github.io/docs/getting-started/netcore/cmdline#write-first-theory)?

Reply:
Yeah!